### PR TITLE
mgr/dashboard: Fix `npm run fixmod` command

### PR DESCRIPTION
**Pull request by p-na**
As the previous implementation didn't work for `spec.ts` files.

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

**Explanation**

`npm run fixmod` does the same as `npm run fix`, but quicker. This is due to it only passing files to `prettier` which actually have been changed (but not yet committed). I use it for quickly fixing frontend files before making a commit.

Revelant for developers only.

*edit*

For the record: `prettier` does not accept a list of files to be processed, just one file or a single pattern, which is hard to implement for "npm scripts", therefore I adapted and replaced this solution to use a new package which does the same thing.

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by s0nea** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
retest this please.

**Comment by s0nea** [comment]
jenkins test dashboard

**Comment by tchaikov** [comment]
retest this please.

**Comment by ricardoasmarques** [comment]
jenkins test make check arm64

**Comment by votdev** [comment]
I tried this PR by running the command without changing a file, but the tool reported 42 changed files. How does this happen?
```
╰─# npm run fixmod

> ceph-dashboard@0.0.0 fixmod /ceph/src/pybind/mgr/dashboard/frontend
> pretty-quick

🔍  Finding changed files since git revision bc49c2ec85.
🎯  Found 42 changed files.
✅  Everything is awesome!
```
After modifying only one JS file the tool is reporting the following:
```
🔍  Finding changed files since git revision bc49c2ec85.
🎯  Found 43 changed files.
✅  Everything is awesome!
```
OK, it noticed the modified file, but where did the others came from?

**Comment by votdev** [comment]
> I tried this PR by running the command without changing a file, but the tool reported 42 changed files. How does this happen?
> 
> ```
> ╰─# npm run fixmod
> 
> > ceph-dashboard@0.0.0 fixmod /ceph/src/pybind/mgr/dashboard/frontend
> > pretty-quick
> 
> 🔍  Finding changed files since git revision bc49c2ec85.
> 🎯  Found 42 changed files.
> ✅  Everything is awesome!
> ```
> 
> After modifying only one JS file the tool is reporting the following:
> 
> ```
> 🔍  Finding changed files since git revision bc49c2ec85.
> 🎯  Found 43 changed files.
> ✅  Everything is awesome!
> ```
> 
> OK, it noticed the modified file, but where did the others came from?

I think these are the compiled JS files that are generated when running ``npm start``, but aren't they ignored by ``.gitignore``?

**Comment by p-na** [comment]
@votdev Thanks for the thorough review! I found out that the default branch `pretty-quick` compares with is `master`. It seems I had an updated master, so that it looked correct for me. Your repository was a few commits behind master. I fixed this. I also added the same pattern to be used to find the files as we use for `prettier`.

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "lint": "npm run lint:tsc && npm run lint:tslint && npm run lint:prettier && npm run lint:html",
     "fix:prettier": "prettier --write \"{src,e2e}/**/*.{ts,scss}\"",
     "fix:tslint": "npm run lint:tslint -- --fix",
-    "fixmod": "prettier --write $(git rev-parse --show-toplevel)/$(git status --porcelain | grep -E '^(\\sM|\\?\\?|A)' | sed -e 's/^...//' | grep -E '^(src|e2e).*\\.(ts|scss)$') 1>/dev/null 2>&1 && echo 'done' || echo 'no changes found'",
+    "fixmod": "pretty-quick --pattern \"{src,e2e}/**/*.{ts,scss}\" --branch HEAD",
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/package.json]: Would it be possible to add the ``--verbose`` know, i would like to see which files are processed.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-09_04:37:10-rados-wip-kefu-testing-2019-06-09-0009-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-11_01:51:33-rados-wip-kefu-testing-2019-06-09-0009-distro-basic-smithi/

test failures tracked by 

- http://tracker.ceph.com/issues/17882
- http://tracker.ceph.com/issues/40120
- the misdirected op issue is not related to this issue.

**Pull request merged**